### PR TITLE
Backport: #2315

### PR DESCRIPTION
**Pull request by sreichel**
### Related Pull Requests

1. See OpenMage/magento-lts#2315

### Contribution checklist (*)
 - [x] Pull request has a meaningful description of its purpose
 - [x] All commits are accompanied by meaningful commit messages
 - [x] All automated tests passed successfully (all builds are green)
 - [x] Add yourself to contributors list

### CODE REVIEW

**Comment by sreichel** [comment]
@elidrissidev when time for it, maybe you can change the code to avoid these phpstan errors?

**Comment by elidrissidev** [comment]
Phpstan errors already solved in a following PR to my original one.

File: app/code/core/Mage/Sales/Model/Api2/Order/Comment/Rest/Admin/V1.php [hunk 1]
```diff
@@ -27,4 +27,67 @@
...
+                Mage::getDesign()->setArea($oldArea);
+            }
+        } catch (Mage_Core_Exception $e) {
+            $this->_critical($e->getMessage(), self::RESOURCE_INTERNAL_ERROR);
+        } catch (Throwable $t) {
+            Mage::logException($t);
+            $this->_critical($t->getMessage(), self::RESOURCE_UNKNOWN_ERROR);
+        }
+
+        return $this->_getLocation($historyItem);
```
Comment (elidrissidev) [inline app/code/core/Mage/Sales/Model/Api2/Order/Comment/Rest/Admin/V1.php:73]: Also I later discovered the original PR had an error at this line after you add a comment. Apparently no one tested it.

Comment (sreichel) [reply app/code/core/Mage/Sales/Model/Api2/Order/Comment/Rest/Admin/V1.php:73]: Mhh, i'd still merge it to have less differences to v20.0.

**Comment by elidrissidev** [comment]
Need to cherry-pick [d91170b](https://github.com/OpenMage/magento-lts/commit/d91170be3e1858c82c8f31cab5adf435333b8f87) for phpstan fix.

**Comment by sreichel** [comment]
Ok, lets merge this and then cherry-pick #2316 

(btw ... i'd set both varibles to null before try/catch to avoid unnecessary calls ... do that later)

**Comment by elidrissidev** [comment]
Why not add them all in the same PR? I already opened a PR to fix the error with `_getLocation()`

**Comment by sreichel** [comment]
@elidrissidev this PR should only backport changes made to v.20.

**Review by fballiano — APPROVED** [review]

**Review by kiatng — APPROVED** [review]

**Pull request merged**